ddtrace/internal/runtime/__init__.py

```diff
@@ -23,38 +25,47 @@ def _generate_runtime_id():
 _RUNTIME_ID = _generate_runtime_id()
 
 
-if hasattr(os, "register_at_fork"):
+def ddtrace_at_fork():
+    global _RUNTIME_ID
+
+    _RUNTIME_ID = _generate_runtime_id()
+    _rand.seed()
 
-    def _set_runtime_id():
-        global _RUNTIME_ID
-        _RUNTIME_ID = _generate_runtime_id()
 
-    os.register_at_fork(after_in_child=_set_runtime_id)
+if hasattr(os, "register_at_fork"):
 
-    def get_runtime_id():
+    def get_runtime_id(pid=None):
         return _RUNTIME_ID
 
+    os.register_at_fork(after_in_child=ddtrace_at_fork)
+
+    AT_FORK_INSTALLED = True
+
 
 else:
     # Non-POSIX systems or Python < 3.7
     import threading
 
     _RUNTIME_PID = os.getpid()
-
-    def _set_runtime_id():
-        global _RUNTIME_ID, _RUNTIME_PID
-        _RUNTIME_ID = _generate_runtime_id()
-        _RUNTIME_PID = os.getpid()
-
     _RUNTIME_LOCK = threading.Lock()
 
-    def get_runtime_id():
-        with _RUNTIME_LOCK:
+    def get_runtime_id(pid=None):
+        if pid is None:
             pid = os.getpid()
+
+        # A lock is required here because it's possible that a process forks
+        # and has a 2 or more threads that will make a call to get_runtime_id()
+        # which will result in a race condition to update _RUNTIME_ID.
```

Comment:
```suggestion
        # A lock is required here because it's possible that a child process
        # has more than one thread that will make a call to get_runtime_id()
        # which will result in a race condition to update _RUNTIME_ID.
```

should be enough